### PR TITLE
drm: pick one primary plane if we have multiple

### LINKED ISSUE #1462: Primary plane selection broken on some devices
It seems we forgot handling devices where the same primary plane can be used on different CRTC. In this case we might end up using the same primary plane multiple times, which apparently can not work.

### PR DESCRIPTION
**Pull request by cmeissl**
some drivers might offer multiple candidates as the primary plane for a CRTC. pick one and claim it so that we do not accidentially select the same plane multiple times.

fixes #1462

### CODE REVIEW

File: anvil/src/udev.rs [hunk 1]
```diff
@@ -523,7 +523,9 @@ impl DrmLeaseHandler for AnvilState<UdevData> {
                 builder.add_connector(conn);
                 builder.add_crtc(*crtc);
                 let planes = backend.drm.planes(crtc).map_err(LeaseRejected::with_cause)?;
-                builder.add_plane(planes.primary.handle);
+                for plane in &planes.primary {
```
Comment (cmeissl) [inline anvil/src/udev.rs:526]: @Drakulix I am not sure if this is correct?

Comment (Drakulix) [reply anvil/src/udev.rs:526]: No it isn't. All resources passed here, are given to the client and I don't think any client has use for multiple primary planes. We should just pick one here (ideally a free one :sweat_smile: ).

**Review by Drakulix — COMMENTED** [review]
Looks good on first glace

File: anvil/src/udev.rs (hunk 1 above)
Comment (cmeissl) [reply anvil/src/udev.rs]: I somewhat expected this to be incorrect :smile: 

Just picking a currently free one might again result in duplicate plane usage. I added the claim logic to the
lease builder. Afaict this should be good?

Comment (Drakulix) [reply anvil/src/udev.rs]: Looks correct now :+1:

**Review by Drakulix — APPROVED** [review]
LGTM!

**Pull request merged**